File: backend/ts/habilidade.ts
import type { Jogador } from './jogador';
import type { Jogo } from './jogo';
import type { Bando } from './bando';

/** Interface base para habilidades de tribo (RF15). */
export abstract class Habilidade {
  abstract executar(jogador: Jogador, jogo: Jogo, bando: Bando): void;
}

/** Habilidade padrão para tribos sem efeito especial definido. */
export class HabilidadeNula extends Habilidade {
  executar(_jogador: Jogador, _jogo: Jogo, _bando: Bando): void {
    // sem efeito
  }
}

File: backend/python/ethnos/habilidade.py
from __future__ import annotations
from abc import ABC, abstractmethod
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .jogador import Jogador
    from .jogo import Jogo
    from .bando import Bando


class Habilidade(ABC):
    """Interface base para habilidades de tribo (RF15)."""

    @abstractmethod
    def executar(self, jogador: "Jogador", jogo: "Jogo", bando: "Bando") -> None:
        pass


class HabilidadeNula(Habilidade):
    """Habilidade padrão para tribos sem efeito especial definido."""

    def executar(self, jogador: "Jogador", jogo: "Jogo", bando: "Bando") -> None:
        pass
